test/TestUtilities/Test.Utility/CommandRunner.cs

```diff
@@ -7,141 +7,125 @@
 using System.Diagnostics;
 using System.IO;
 using System.Text;
+using System.Threading;
 
 namespace NuGet.Test.Utility
 {
     public class CommandRunner
     {
+        private static readonly SemaphoreSlim Semaphore = new SemaphoreSlim(1, 1);
```

Comment:
Would it be "better" to change the relevant test projects xunit settings to not allow tests in that project to be run in parallel?

Our tests are already so slow, with these integration tests being the slowest of the slow, so it seems a real shame to limit them to 1 test at a time.

Having said that, xunit runs tests within a collection serially, and only runs different collections in parallel (each class is its own collection if no collection attribute is explicitly set). However, since all dotnet integration tests use the same collection, in order to get access to the singleton test fixture, all dotnet.integration.tests should already be running sequentially, so this semaphore shouldn't change any behaviour in that test project. I can't remember if NuGet.CommandLine.Tests use a collection, but if it doesn't, then this semaphore will make those tests significantly slower, when my understanding is you're trying to target dotnet.integration.tests.

Reply:
Excellent question.  What bothers me the most is I never found a smoking gun as to why `dotnet` was crashing on Mac for .NET 8.  I added this as a big hammer to see if it would fix the problem (along with a few other improvements) and things started working better.  I'll push a commit with the semaphore removed and see it works.  Otherwise, I can move the semaphore to the `DotnetIntegrationTestFixture` instead so it doesn't affect other tests.